dev/bots/analyze.dart

```diff
@@ -472,6 +473,36 @@ Future<void> _verifyNoBadImportsInFlutterTools(String workingDirectory) async {
   }
 }
 
+final RegExp _licensePattern = RegExp(r'''LICENSE file''');
+
+Future<void> _verifyNoMissingLicense(String workingDirectory) async {
+  final List<String> errors = <String>[];
+  for (FileSystemEntity entity in Directory(path.join(workingDirectory, 'packages'))
+      .listSync(recursive: true)
+      .where((FileSystemEntity entity) => entity is File && path.extension(entity.path) == '.dart')) {
+    final File file = entity;
+    bool hasLicense = false;
+    for (String line in file.readAsLinesSync()) {
```

Comment:
Don't we want to enforce that the license header is at the beginning of the file?

Reply:
I guess so. Is it safe to assume that `LICENSE file` should appear in the top 10 lines of the file?

Reply:
Would it be problematic to just expect that the first line of the file starts with "Copyright \d{4}"?

Reply:
Good idea. Done 😄